crates/pessimistic-proof/src/batch_header.rs

```diff
@@ -0,0 +1,133 @@
+#![allow(clippy::too_many_arguments)]
+use std::{collections::BTreeMap, hash::Hash};
+
+use reth_primitives::U256;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_with::serde_as;
+
+use crate::{
+    bridge_exit::{BridgeExit, NetworkId, TokenInfo},
+    imported_bridge_exit::ImportedBridgeExit,
+    local_balance_tree::LocalBalancePath,
+    local_exit_tree::hasher::Hasher,
+    nullifier_tree::NullifierPath,
+};
+
+/// Represents the data submitted by the CDKs to the AggLayer.
+///
+/// The bridge exits plus the imported bridge exits define
+/// the state transition, resp. the amount that goes out and the amount that comes in.
+///
+/// The bridge exits refer to the [`BridgeExit`]  emitted by
+/// the origin network of the [`BatchHeader`].
+///
+/// The imported bridge exits refer to the [`BridgeExit`] received and imported
+/// by the origin network of the [`BatchHeader].
+#[serde_as]
+#[derive(Default, Clone, Debug, Serialize, Deserialize)]
+pub struct BatchHeader<H>
+where
+    H: Hasher,
+    H::Digest: Eq + Hash + Copy + Serialize + DeserializeOwned,
+{
+    /// The origin network which emitted this BatchHeader.
+    /// TODO: we should clarify naming. We use origin to refer to the issuing network of a token. Could consider using "sending" or "local" to refer to the network that created a batch.
+    pub origin_network: NetworkId,
+
+    // /// The start slot height and end block height for the batch. Unsure if required
+    // pub start_block_height: u32,
+    // pub end_block_height u32,
+
+    // /// The state root produced by the current block
+    // TODO: Determine if this is necessary
```

Comment:
not necessary I think

Reply:
Definitely not necessary for PP generation, but if the same header is used for execution and consensus proofs (which it should be) then it could be useful to include. Happy to omit for now though.